src/opentimelineio/clip.cpp

```diff
@@ -19,33 +25,74 @@ Clip::~Clip()
 MediaReference*
 Clip::media_reference() const noexcept
 {
-    return _media_reference;
+    auto active = _media_references.find(_active_media_reference);
+    return active == _media_references.end() || !active->second
+               ? nullptr
+               : active->second;
+}
+
+Clip::MediaReferences
+Clip::media_references() const noexcept
+{
+    MediaReferences result;
+    for (auto const& m: _media_references)
+    {
+        result.insert(
+            { m.first, dynamic_retainer_cast<MediaReference>(m.second) });
+    }
+    return result;
+}
+
+void
+Clip::set_media_references(MediaReferences const& media_references) noexcept
+{
+    _media_references.clear();
+    for (auto const& m: media_references)
+    {
+        _media_references[m.first] = m.second;
+    }
+}
+
+std::string
+Clip::active_media_reference() const noexcept
+{
+    return _active_media_reference;
+}
+
+void
+Clip::set_active_media_reference(std::string const& new_active) noexcept
```

Comment:
The corollary to my comment above is that someone can also set a key here that does not match any in the map, so whatever error handling we decide on we should do the same here.